### PR TITLE
Calling 3rd party APIs

### PR DESCRIPTION
**Pull request by zoefarrell**

### CODE REVIEW

File: module5/preparation/Week4/CallingThirdPartyAPIs.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+### Research
+At the start of tomorrow's lesson we are also going to talk briefly about asynchronous vs synchronous code and the keywords `Async` and `Await`.
+
+To prepare for that discussion, research the following questions and be prepared to share your answers:
+
+* What is the difference between synchronous and asynchronous code?
+* What do the keywords `Async` and `Await` do in C#?
+
+✅ Deliverable: Share one of the resources you found helpful in learning about these topics on the Slack thread your instructor will create.
```
Comment (memcmahon) [inline module5/preparation/Week4/CallingThirdPartyAPIs.md:17]: Love this deliverable!

File: module5/lessons/Week4/CallingThirdPartyAPIs.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+## Learning Goals
+* Build familiarity with asynchronous code and Async/Await
+* Understand how to make a GET request to a Third Party API
+* Learn how to make a POST request
+
+### Overview
+
+In Mod 4 you were introduced to Third Party APIs and got a ton of practice calling various APIs from Postman and reading documentation. Maybe you remember working with the Github API, The Pokemon API, and the Cat API.
+
+Today we're going to take the exciting leap of calling APIs, such as those listed above, from your applications! You will learn how to retrieve data from the API and then display it to the user.
```
Comment (memcmahon) [inline module5/lessons/Week4/CallingThirdPartyAPIs.md:15]: This is going to be so cool - some students have already played around with this in the Mod4 project!!

File: module5/lessons/Week4/CallingThirdPartyAPIs.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+
+### Learning From a Tutorial
+Follow [this](https://www.ezzylearning.net/tutorial/how-to-consume-third-party-web-apis-in-asp-net-core) tutorial individually up until the section **Managing HttpClient objects with IHttpClientFactory**. You will have roughly 1.5 hours, which should be enough time to thoughtfully work through it.
+
+As you work through the tutorial, add comments to specify what each part of your code is doing. I also highly recommend that you type the code instead of copy/pasting for topics that feel new.
+
+As with all tutorials, you have to expect it might not work perfectly and you might need to do some debugging to get your code to work! If you're stuck on something for more than 5 or 10 minutes reach out to your peers in code-help!
+
+Deliverable✅: Once you have finished the tutorial, send a message to your instructor with your answer to the following question:
```
Comment (memcmahon) [inline module5/lessons/Week4/CallingThirdPartyAPIs.md]: I wonder if we should also have a deliverable question that is related to the tutorial outcome? Something to put some pressure on to dig in and get through the tutorial fully

Comment (memcmahon) [reply module5/lessons/Week4/CallingThirdPartyAPIs.md]: Ope - now I see the other questions below :)

File: module5/labs/Week4/CallingThirdPartyAPIs.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+---
+layout: page
+title: Calling Third Party APIs Lab
+---
+
+### Overview
```
Comment (memcmahon) [inline module5/labs/Week4/CallingThirdPartyAPIs.md:6]: Love this lab!

**Review by memcmahon — APPROVED** [review]

File: module5/preparation/Week4/CallingThirdPartyAPIs.md (hunk 1 above)
Comment (zoefarrell) [reply module5/preparation/Week4/CallingThirdPartyAPIs.md:17]: Sometimes great ideas come out of being time crunched 🙃

File: module5/lessons/Week4/CallingThirdPartyAPIs.md (hunk 3 above)
Comment (zoefarrell) [reply module5/lessons/Week4/CallingThirdPartyAPIs.md]: I waws thinking about this too. I wish the students were all able to self motivate at this point and keep pushing forward through the deliverable, but might be best to have a "did you do the thing?" part of the deliverable too. Maybe a screenshot of searching for a country outside of the US.

File: module5/labs/Week4/CallingThirdPartyAPIs.md (hunk 4 above)
Comment (zoefarrell) [reply module5/labs/Week4/CallingThirdPartyAPIs.md:6]: Thanks! Glad we brainstormed together, because I'm very excited about this one too!!

**Pull request merged**